middle_end/flambda2/simplify/simplify_binary_primitive.ml

```diff
@@ -967,6 +967,31 @@ let simplify_array_load (array_kind : P.Array_kind.t) mutability dacc
     let dacc = DA.add_variable dacc result_var ty in
     SPR.create named ~try_reify:false dacc
 
+let simplify_array_vector_load (vec_kind : Lambda.boxed_vector)
```

Comment:
Not sure if this function is necessary, since vector loads should only ever operate on monomorphic arrays.

Reply:
well even in that case, it can be useful to remove dead code branches based on the `Bottom` case.